### PR TITLE
Fix stream processor dependencies

### LINKED ISSUE #3339: BrokerReprocessingTest is flaky
**Description**

The BrokerReprocessingTest fails from time to time with:

```
Error
Deployment partition not installed into the container within 15 SECONDS.
Stacktrace
java.lang.RuntimeException: Deployment partition not installed into the container within 15 SECONDS.
	at io.zeebe.broker.test.EmbeddedBrokerRule.startBroker(EmbeddedBrokerRule.java:238)
	at io.zeebe.broker.it.startup.BrokerReprocessingTest.restartBroker(BrokerReprocessingTest.java:705)
	at io.zeebe.broker.it.startup.BrokerReprocessingTest.shouldAssignUniqueWorkflowInstanceKeyAfterRestart(BrokerReprocessingTest.java:487)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.concurrent.TimeoutException: Timeout after: 15 SECONDS
	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:131)
	at io.zeebe.broker.test.EmbeddedBrokerRule.startBroker(EmbeddedBrokerRule.java:233)
	... 18 more
```

This is also reproducible locally. The real error lies in the broker stopping, see the following log.

```

13:47:28.930 [partition-1-processor-snapshot-director] [0.0.0.0:28802-zb-actors-1] DEBUG io.zeebe.logstreams.snapshot - The position of the last valid snapshot is '-1'. Taking snapshots beyond this position.
13:47:36.006 [] [Time-limited test] INFO  io.zeebe.broker.system - Broker shutting down...
13:47:36.209 [exporter-1] [0.0.0.0:28802-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Closed exporter director 'exporter-1'.
13:47:36.521 [service-container-main] [0.0.0.0:28802-zb-actors-0] DEBUG io.zeebe.servicecontainer - Removed service broker.httpServer
13:47:36.611 [] [zb-blocking-task-runner-6-0.0.0.0:28802] DEBUG io.zeebe.gateway - Closing gateway broker client ...
13:47:36.620 [] [zb-blocking-task-runner-6-0.0.0.0:28802] DEBUG io.zeebe.gateway - topology manager closed
13:47:36.819 [] [zb-blocking-task-runner-6-0.0.0.0:28802] DEBUG io.zeebe.gateway - transport closed
13:47:36.820 [] [zb-blocking-task-runner-6-0.0.0.0:28802] DEBUG io.zeebe.gateway - data frame receive buffer closed
13:47:36.909 [] [zb-blocking-task-runner-6-0.0.0.0:28802] DEBUG io.zeebe.gateway - Gateway broker client closed.
13:47:56.010 [] [Time-limited test] DEBUG io.zeebe.servicecontainer - Service container closing failed. Print dependencies.
13:47:56.013 [] [Time-limited test] DEBUG io.zeebe.servicecontainer - 
logstream.raft-atomix-partition-1.service in AwaitDependentsStopped
	\
 	-- logstream.raft-atomix-partition-1 in AwaitDependentsStopped
 	-- logstream.raft-atomix-partition-1.storage in AwaitDependentsStopped
transport.commandApi.server.server in AwaitDependentsStopped
	\
cluster.base.bootstrapped in AwaitDependentsStopped
	\
cluster.base.atomix.join in AwaitDependentsStopped
	\
 	-- cluster.base.atomix in AwaitDependentsStopped
 	-- cluster.base.topologyManager in AwaitDependentsStopped
cluster.base.partition.raft-atomix-partition-1.leader in AwaitDependentsStopped
	\
 	-- logstream.raft-atomix-partition-1.service in AwaitDependentsStopped
 	-- cluster.base.partition.leader.install.root.raft-atomix-partition-1 in AwaitDependentsStopped
 	-- raft.leader.raft-atomix-partition-1.openLogStream in AwaitDependentsStopped
broker.system.management.requestHandler in AwaitDependentsStopped
	\
 	-- cluster.base.atomix in AwaitDependentsStopped
cluster.base.atomix in AwaitDependentsStopped
	\
 	-- cluster.base.bootstrapped in AwaitDependentsStopped
 	-- cluster.base.raft.configurationManager in AwaitDependentsStopped
cluster.base.topologyManager in AwaitDependentsStopped
	\
 	-- cluster.base.atomix in AwaitDependentsStopped
 	-- cluster.base.bootstrapped in AwaitDependentsStopped
cluster.base.partition.leader.install.root.raft-atomix-partition-1 in AwaitDependentsStopped
	\
 	-- cluster.base.partition.install.raft-atomix-partition-1 in AwaitDependentsStopped
cluster.base.partitions.bootstrap in AwaitDependentsStopped
	\
 	-- cluster.base.atomix in AwaitDependentsStopped
 	-- cluster.base.raft.configurationManager in AwaitDependentsStopped
 	-- cluster.base.atomix.join in AwaitDependentsStopped
cluster.base.raft.configurationManager in AwaitDependentsStopped
	\
 	-- cluster.base.bootstrapped in AwaitDependentsStopped
logstream.raft-atomix-partition-1 in AwaitDependentsStopped
	\
raft.leader.raft-atomix-partition-1.openLogStream in AwaitDependentsStopped
	\
 	-- logstream.raft-atomix-partition-1.distributed.log in AwaitDependentsStopped
 	-- logstream.raft-atomix-partition-1.service in AwaitDependentsStopped
 	-- cluster.base.partition.leader.install.root.raft-atomix-partition-1 in AwaitDependentsStopped
logstream.raft-atomix-partition-1.storage in AwaitDependentsStopped
	\
 	-- logstream.raft-atomix-partition-1 in AwaitDependentsStopped
cluster.base.partition.install.raft-atomix-partition-1 in AwaitDependentsStopped
	\
 	-- cluster.base.partitions.bootstrap in AwaitDependentsStopped
logstream.raft-atomix-partition-1.distributed.log in AwaitDependentsStopped
	\
 	-- cluster.base.atomix in AwaitDependentsStopped
 	-- cluster.base.partition.leader.install.root.raft-atomix-partition-1 in AwaitDependentsStopped
logstream.raft-atomix-partition-1.stream-processor in AwaitDependentsStopped
	\
 	-- cluster.base.atomix in AwaitDependentsStopped
 	-- cluster.base.topologyManager in AwaitDependentsStopped
 	-- cluster.base.partition.leader.install.root.raft-atomix-partition-1 in AwaitDependentsStopped
 	-- transport.commandApi.messageHandler in AwaitDependentsStopped
 	-- cluster.base.partition.raft-atomix-partition-1.leader in AwaitDependentsStopped
 	-- broker.system.management.requestHandler in AwaitDependentsStopped
transport.commandApi.messageHandler in AwaitDependentsStopped
	\
 	-- transport.commandApi.server.server in AwaitDependentsStopped
logstream.raft-atomix-partition-1.snapshot-director in AwaitStopState
	\
 	-- logstream.raft-atomix-partition-1.stream-processor in AwaitDependentsStopped
 	-- cluster.base.partition.leader.install.root.raft-atomix-partition-1 in AwaitDependentsStopped
 	-- cluster.base.partition.raft-atomix-partition-1.leader in AwaitDependentsStopped
13:47:56.016 [] [Time-limited test] ERROR io.zeebe.broker.system - Failed to close broker within PT20S seconds.
java.util.concurrent.TimeoutException: Timeout after: 20000 MILLISECONDS
	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:131) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.servicecontainer.impl.ServiceContainerImpl.close(ServiceContainerImpl.java:117) ~[zeebe-service-container-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.system.SystemContext.close(SystemContext.java:175) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.lambda$close$1(Broker.java:71) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.util.LogUtil.doWithMDC(LogUtil.java:21) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.close(Broker.java:67) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.test.EmbeddedBrokerRule.stopBroker(EmbeddedBrokerRule.java:196) ~[zeebe-broker-0.22.0-SNAPSHOT-tests.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.it.startup.BrokerReprocessingTest.restartBroker(BrokerReprocessingTest.java:701) ~[test-classes/:?]
	at io.zeebe.broker.it.startup.BrokerReprocessingTest.shouldAssignUniqueWorkflowInstanceKeyAfterRestart(BrokerReprocessingTest.java:487) [test-classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit-4.12.jar:4.12]
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239) [junit-4.12.jar:4.12]
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48) [junit-4.12.jar:4.12]
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55) [junit-4.12.jar:4.12]
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298) [junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292) [junit-4.12.jar:4.12]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
13:47:56.204 [] [Time-limited test] INFO  io.zeebe.broker.system - Broker shut down.
13:47:56.615 [] [Time-limited test] DEBUG io.zeebe.broker.system - Initializing configuration with base path /tmp/3d719665-5291-4a0f-8930-f6e004831d36
13:47:56.712 [] [Time-limited test] INFO  io.zeebe.broker.system - Version: 0.22.0-SNAPSHOT
13:47:56.800 [] [Time-limited test] INFO  io.zeebe.broker.system - Starting broker with configuration {
  "network": {
    "host": "0.0.0.0",
    "portOffset": 0,
    "maxMessageSize": "4M",
    "commandApi": {
      "defaultPort": 26501,
      "host": "0.0.0.0",
      "port": 28802,
      "advertisedHost": "0.0.0.0",
      "advertisedPort": 28802
    },
    "internalApi": {
      "defaultPort": 26502,
      "host": "0.0.0.0",
      "port": 28803,
      "advertisedHost": "0.0.0.0",
      "advertisedPort": 28803
    },
    "monitoringApi": {
      "defaultPort": 9600,
      "host": "0.0.0.0",
      "port": 28804,
      "advertisedHost": "0.0.0.0",
      "advertisedPort": 28804
    }
  },
  "cluster": {
    "initialContactPoints": [],
    "partitionIds": [
      1
    ],
    "nodeId": 0,
    "partitionsCount": 1,
    "replicationFactor": 1,
    "clusterSize": 1,
    "clusterName": "zeebe-cluster",
    "gossipFailureTimeout": 10000,
    "gossipInterval": 250,
    "gossipProbeInterval": 1000
  },
  "threads": {
    "cpuThreadCount": 2,
    "ioThreadCount": 2
  },
  "data": {
    "directories": [
      "/tmp/3d719665-5291-4a0f-8930-f6e004831d36/data"
    ],
    "logSegmentSize": "512M",
    "snapshotPeriod": "15m",
    "raftSegmentSize": "512M",
    "maxSnapshots": 3
  },
  "exporters": [
    {
      "id": "test-recorder",
      "className": "io.zeebe.test.util.record.RecordingExporter"
    }
  ],
  "gateway": {
    "enable": true,
    "network": {
      "host": "0.0.0.0",
      "port": 28800
    },
    "cluster": {
      "contactPoint": "0.0.0.0:28803",
      "maxMessageSize": "4M",
      "requestTimeout": "15s",
      "clusterName": "zeebe-cluster",
      "memberId": "gateway",
      "host": "0.0.0.0",
      "port": 28801
    },
    "threads": {
      "managementThreads": 1
    },
    "monitoring": {
      "enabled": false,
      "host": "0.0.0.0",
      "port": 9600
    },
    "security": {
      "enabled": false
    }
  },
  "backpressure": {
    "enabled": true,
    "useWindowed": true,
    "algorithm": "vegas"
  }
}
13:47:56.817 [service-controller] [0.0.0.0:28802-zb-actors-0] INFO  io.zeebe.transport - Bound commandApi.server to /0.0.0.0:28802
13:47:57.100 [] [zb-blocking-task-runner-1-0.0.0.0:28802] INFO  io.zeebe.gateway - Version: 0.22.0-SNAPSHOT
13:47:57.103 [] [zb-blocking-task-runner-1-0.0.0.0:28802] INFO  io.zeebe.gateway - Starting gateway with configuration {
  "enable": true,
  "network": {
    "host": "0.0.0.0",
    "port": 28800
  },
  "cluster": {
    "contactPoint": "0.0.0.0:28803",
    "maxMessageSize": "4M",
    "requestTimeout": "15s",
    "clusterName": "zeebe-cluster",
    "memberId": "gateway",
    "host": "0.0.0.0",
    "port": 28801
  },
  "threads": {
    "managementThreads": 1
  },
  "monitoring": {
    "enabled": false,
    "host": "0.0.0.0",
    "port": 9600
  },
  "security": {
    "enabled": false
  }
}
13:47:57.103 [service-controller] [0.0.0.0:28802-zb-actors-1] DEBUG io.zeebe.broker.clustering - Adding Node{nodeId=0, commandApi=0.0.0.0:28802} to list of known members
13:47:57.620 [] [netty-messaging-event-epoll-server-0] WARN  io.atomix.cluster.messaging.impl.NettyMessagingService - Failed to bind TCP server to port 0.0.0.0:28803 due to io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
13:47:57.622 [service-controller] [0.0.0.0:28802-zb-actors-0] ERROR io.zeebe.servicecontainer - Service cluster.base.atomix.join failed to start while in AwaitStartState
java.util.concurrent.CompletionException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:777) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) ~[?:?]
	at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$bind$31(NettyMessagingService.java:577) ~[atomix-cluster-3.2.0-SNAPSHOT.jar:?]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:500) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:474) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:413) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:538) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:531) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:111) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetFailure(AbstractChannel.java:991) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:553) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1345) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:503) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:488) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:984) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:247) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:355) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:331) ~[netty-transport-native-epoll-4.1.39.Final-linux-x86_64.jar:4.1.39.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at java.lang.Thread.run(Thread.java:834) ~[?:?]
Caused by: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
13:47:57.809 [service-container-main] [0.0.0.0:28802-zb-actors-1] ERROR io.zeebe.servicecontainer - Failed to build service
java.util.concurrent.CompletionException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:777) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) ~[?:?]
	at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$bind$31(NettyMessagingService.java:577) ~[atomix-cluster-3.2.0-SNAPSHOT.jar:?]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:500) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:474) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:413) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:538) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:531) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:111) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetFailure(AbstractChannel.java:991) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:553) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1345) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:503) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:488) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:984) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:247) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:355) ~[netty-transport-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:331) ~[netty-transport-native-epoll-4.1.39.Final-linux-x86_64.jar:4.1.39.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.39.Final.jar:4.1.39.Final]
	at java.lang.Thread.run(Thread.java:834) ~[?:?]
Caused by: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
13:48:12.808 [] [Time-limited test] INFO  io.zeebe.broker.system - Broker shutting down...
```

BrokerReprocessingTest

A clear and concise description of what this issue is about.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

* processor and exporter used the logstorage and writer from the logstream, but didn't depend on them
 * to avoid this usage of the log stream the reader and writer need to be created directly with the buffer and log storage to enforce the thinking about the current usage

This is the new dependency graph

fixed

## Related issues

closes #3339 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: logstreams/src/main/java/io/zeebe/logstreams/log/LogStreamReader.java [hunk 1]
```diff
@@ -34,15 +35,15 @@
    *
    * @param log the stream which provides the log
```
Comment (saig0) [inline logstreams/src/main/java/io/zeebe/logstreams/log/LogStreamReader.java]:
```suggestion
   * @param log the log storage to read from
```

File: logstreams/src/main/java/io/zeebe/logstreams/log/LogStreamReader.java [hunk 2]
```diff
@@ -34,15 +35,15 @@
...
    */
-  void wrap(LogStream log);
+  void wrap(LogStorage log);
 
   /**
    * Initialize the reader and seek to the given log position.
    *
    * @param log the stream which provides the log
```
Comment (saig0) [inline logstreams/src/main/java/io/zeebe/logstreams/log/LogStreamReader.java]:
```suggestion
   * @param log the log storage to read from
```

File: logstreams/src/test/java/io/zeebe/distributedlog/DistributedLogPartitionRule.java [hunk 3]
```diff
@@ -68,15 +69,19 @@ private void getLogStream() {
     final ServiceName<Void> testService =
         ServiceName.newServiceName(String.format("test-%s", logName), Void.class);
     final Injector<LogStream> logStreamInjector = new Injector<>();
+    final Injector<LogStorage> logStorageInjector = new Injector<>();
     serviceContainer
         .createService(testService, () -> null)
         .dependency(logStreamServiceName(logName), logStreamInjector)
+        //        .dependency(logStorageServiceName(logName), logStorageInjector)
```
Comment (saig0) [inline logstreams/src/test/java/io/zeebe/distributedlog/DistributedLogPartitionRule.java]: Do. Or not do. There is no uncommented code.

**Review by saig0 — APPROVED** [review]
:+1: 

Just some cleanups.

**Comment by saig0** [comment]
Looking at the dependency graph, I'm wondering why the `logsteam_*_deletion` service doesn't have a dependency to the log-stream :confused:

**Comment by Zelldon** [comment]
No more transitive due to the exporter and partition 😁😅

**Pull request merged**